src/components/UserProfileSettings/AddEmail.tsx

```diff
@@ -0,0 +1,263 @@
+  }, [userPushSDKInstance, handleConnectWalletAndEnableProfile, wallet]);
+
+  // Watch for code length and submit automatically if it reaches 6 digits
+  useEffect(() => {
+    if (codeFormik.values.code.length === 6 && !codeFormik.errors.code) {
+      codeFormik.submitForm();
+    }
+  }, [codeFormik.values.code, codeFormik.errors.code]);
+
+  const handleSendVerificationCode = async () => {
+    const sdkInstance = await getSDKInstance();
+    const data = {
+      wallet: caip10WalletAddress,
+      value: emailFormik.values.email,
+      valueType: 'email',
+    };
+
+    const verificationProof = await generateVerificationProof(data, sdkInstance);
+
+    if (!verificationProof) return;
+
+    sendVerification(
+      {
+        caipAddress: caip10WalletAddress as string,
+        verificationProof: verificationProof as string,
+        value: emailFormik.values.email,
+        social_platform: 'email',
+      },
+      {
+        onSuccess: (response: any) => {
+          if (response?.success) {
+            setStep(Steps.VerifyCode);
+          } else {
+            emailFormik?.setFieldError('email', 'Error sending code. Please try again');
+          }
+        },
+        onError: (error: Error) => {
+          console.log('Error sending code', error);
+        },
+      }
+    );
+  };
+
+  const handleVerificationCode = async () => {
+    const sdkInstance = await getSDKInstance();
+    const data = {
+      wallet: caip10WalletAddress,
+      value: emailFormik.values.email,
+      valueType: 'email',
+      verificationCode: codeFormik.values.code,
+    };
+
+    const verificationProof = await generateVerificationProof(data, sdkInstance);
+
+    if (!verificationProof) return;
+
+    verifyVerification(
+      {
+        caipAddress: caip10WalletAddress as string,
+        verificationCode: codeFormik.values.code,
+        value: emailFormik.values.email,
+        social_platform: 'email',
+      },
+      {
+        onSuccess: (response: any) => {
+          if (response?.success) {
+            onClose();
+            refetchSocialHandleStatus();
+            setSuccessMessage('Email Account was linked successfully');
+          } else {
+            codeFormik?.setFieldError('code', 'Error verifying code. Please try again');
+          }
+        },
+        onError: (error: Error) => {
+          console.log('Error verifying code', error);
+          setErrorMessage('Error verifying code');
+        },
+      }
+    );
+  };
+
+  const resendVerificationCode = () => {
+    handleSendVerificationCode();
+  };
+
+  return (
+    <Modal
+      size="small"
+      isOpen={isOpen}
+      onClose={onClose}
+      acceptButtonProps={
+        step === Steps.EnterEmail
+          ? {
+              children: 'Next',
+              loading: isSendingVerification,
+              onClick: () => {
+                emailFormik.handleSubmit();
+              },
+            }
+          : null
+      }
```

Comment:
the button should be in loading state when we are performing any action. We can pass a loading prop to the acceptButtonProps button config.